packages/perseus/src/renderer.tsx

```diff
@@ -594,6 +594,24 @@ class Renderer
         return null;
     };
 
+    _getWidgetIndexById(id: string) {
+        const widgetIndex = Object.keys(this.props.widgets).indexOf(id);
```

Comment:
The order of the keys returned by `Object.keys` is the order in which the keys were originally inserted into the object. Is that actually what you want here? I wouldn't want to depend on the key order being the same across renders.

I'd much rather use the order in which the widgets appear in the markdown.

Reply:
I switched to using `this.widgetIds` which seems to be the used IDs from the parsing process.